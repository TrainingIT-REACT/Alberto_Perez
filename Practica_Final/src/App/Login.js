import React from 'react';
import { connect } from 'react-redux';
// Acciones
import { updateName } from './actions/user';

class Login extends React.Component {
  constructor(props) {
    super(props);
    // Creamos la referencia para el input
    this.input = React.createRef();
    // Bind del método
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(e) {
    e.preventDefault();
    this.props.updateName(this.input.current.value);
    this.input.current.value = '';
    this.props.history.push(`/myplaylist`);
  }


  render() {
    return (
      <div className="Login">
        <h2>Login</h2>
        <div className="row">
          <div className="six columns">
            <form onSubmit={this.onSubmit}>
              <div className="row">
                <label htmlFor="name">Name </label>
                <input id="name" type="text" ref={this.input} placeholder="Ángel, María, Cris, etc."/>
              </div>
              <br/>
              <button>Login</button>
            </form>
          </div>
        </div>
      </div>
    );
  }
}


const mapStateToProps = (state) => {
  return {
    name: state.user.name
  }
}

const mapDispatchToProps = (dispatch) => ({
  updateName: (name) => dispatch(updateName(name)),
});


export default connect(mapStateToProps, mapDispatchToProps)(Login);
