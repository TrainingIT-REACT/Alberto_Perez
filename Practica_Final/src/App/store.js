import { createStore, combineReducers, applyMiddleware } from "redux";
import thunk from 'redux-thunk';
//import promise from 'redux-promise-middleware';

// Reducers
import album from './reducers/album';
import song from './reducers/song';
import user from './reducers/user';

export default createStore(combineReducers({ user, album, song }), applyMiddleware(thunk));
