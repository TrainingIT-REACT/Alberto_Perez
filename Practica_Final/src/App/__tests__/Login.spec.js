import React from 'react';
import { render } from 'enzyme';

import { Provider } from "react-redux";
import store from '../store';

// Componente
import Login from '../Login';


describe("Login", () => {
  let wrapper;
  // Inicializamos el componente en un beforeEach para
  // evitar tener que repetir esta lÃ­nea en cada test
  beforeEach(() => {
    wrapper = render(<Provider store={store}><Login/> </Provider>);
  });


  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });


  it('should add the HTML elements', () => {      
    // Comprobamos los distintos aspectos de HTML      
    expect(wrapper.find('form')).toBeTruthy();
    expect(wrapper.find('label').length).toBe(1);
    expect(wrapper.find('input[type="text"]').length).toBe(1);
    expect(wrapper.find('button').length).toBe(1);
  }); 

})


