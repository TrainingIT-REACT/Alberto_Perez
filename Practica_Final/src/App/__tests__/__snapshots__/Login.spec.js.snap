// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
Array [
  <div
    class="Login"
  >
    <h2>
      Login
    </h2>
    <div
      class="row"
    >
      <div
        class="six columns"
      >
        <form>
          <div
            class="row"
          >
            <label
              for="name"
            >
              Name 
            </label>
            <input
              id="name"
              placeholder="Ángel, María, Cris, etc."
              type="text"
            />
          </div>
          <br />
          <button>
            Login
          </button>
        </form>
      </div>
    </div>
  </div>,
  " ",
]
`;
